extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -273,18 +276,33 @@ class MongoQueueProcessor {
      */
     _updateLocations(entry, location) {
         const locations = entry.getLocation();
-        const editLocations = locations.map(l => {
-            const newValues = {
+        if (!locations || locations.length === 0) {
+            const dataStoreVersionId = entry.getVersionId() ?
+                entry.getEncodedVersionId() : '';
+            const editLocation = [{
                 key: entry.getObjectKey(),
+                size: 0,
+                start: 0,
                 dataStoreName: location,
                 dataStoreType: 'aws_s3',
-            };
-            if (entry.getVersionId()) {
-                newValues.dataStoreVersionId = entry.getEncodedVersionId();
-            }
-            return Object.assign({}, l, newValues);
-        });
-        entry.setLocation(editLocations);
+                dataStoreETag: `1:${ZERO_BYTE_ETAG}`,
+                dataStoreVersionId,
+            }];
+            entry.setLocation(editLocation);
+        } else {
+            const editLocations = locations.map(l => {
+                const newValues = {
+                    key: entry.getObjectKey(),
+                    dataStoreName: location,
+                    dataStoreType: 'aws_s3',
+                };
+                if (entry.getVersionId()) {
+                    newValues.dataStoreVersionId = entry.getEncodedVersionId();
+                }
+                return Object.assign({}, l, newValues);
```

Comment:
Minor: `l` seems a bit too vague here. Seems to only add benefit by spelling out `location`.